### PR TITLE
[docs] adds bundling docs and discusses dependency mismatches

### LINKED ISSUE #60: Add section about dependency mismatches
Explain what will happen when there are dependency mismatches between different engines and/or their host app.

### PR DESCRIPTION
**Pull request by gabrielcsapo**
*Why?*

I think it is important to discuss how engines works so that way when engineers are deciding how to structure their engines it will be obvious of what the outcomes should be.

Screen Shot 2020-04-09 at 8 44 17 PM

fixes #60

### CODE REVIEW

**Comment by gabrielcsapo** [comment]
This can be converted to use a table after this lands https://github.com/ember-learn/ember-cli-addon-docs/pull/469

**Review by villander — APPROVED** [review]
Thanks a lot, good information for us.

File: tests/dummy/app/templates/docs/bundling.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Bundling
+
+_Ember Engines_ not only provides isolation principles for how your code is run, but also how it is bundled.
+
+## Host / Engine Dependencies
+
+In the following example, if the dependency required by the _Engine A_ is different than the _Host_ it will resolve the top level dependency. In this case `Foo` will resolve to `Foo@2`. If that dependency is not in the host and only in the Engine, it will be bundled in `engine-vendor-{hash}.js`.
```
Comment (villander) [inline tests/dummy/app/templates/docs/bundling.md]: what happens when the `v2` is in host? the `Foo` is not bundled in the engine, improving de bundle size?

Comment (villander) [reply tests/dummy/app/templates/docs/bundling.md]: Could be awesome describe the other scenario

Comment (gabrielcsapo) [reply tests/dummy/app/templates/docs/bundling.md]: Good idea, adding this now

**Pull request merged**